test-data/unit/check-literal.test

```diff
@@ -2342,12 +2342,12 @@ reveal_type(test.get(good_keys, 3))               # N: Revealed type is 'Union[_
 reveal_type(test.pop(optional_keys))              # N: Revealed type is 'Union[__main__.D, __main__.E]'
 reveal_type(test.pop(optional_keys, 3))           # N: Revealed type is 'Union[__main__.D, __main__.E, Literal[3]?]'
 reveal_type(test.setdefault(good_keys, AAndB()))  # N: Revealed type is 'Union[__main__.A, __main__.B]'
+reveal_type(test.get(bad_keys))                   # N: Revealed type is 'builtins.object*'
+reveal_type(test.get(bad_keys, 3))                # N: Revealed type is 'builtins.object'
```

Comment:
It would be great if a maintainer could have a look at this.
I don't know the difference between `builtins.object` and `builtins.object*` and couldn't find any reference to the asterisk in the documentation.

Therefor I just placed the needed asterisks by trial and error. This might or might not be wrong.

Reply:
The asterisk means that it's an inferred type; I think it can be safely ignored.

I'm not sure why mypy infers `object` here though, and it doesn't seem very useful.

Reply:
According to PEP 589 "The static type of d.get(e) should be object if the string value of e cannot be determined statically".

As far as I can tell PEP 589 doesn't say anything about `d.get(e, default)`.

Also something else not related to my code seems to convert Union[..., builtins.object, ...] to just builtins.object, which can bee seen when making defauly.py line 234 retun Any instead. In that case mypy infers `Union[__main__.A, Literal[3]?, Any]`.

Reply:
The default doesn't change anything, since we can't infer the type of the value from the default (the value type could be an arbitrary union type, for example). 

Since typed dicts are heterogeneous, we can't predict the type of arbitrary keys. In this case the only possible precise type is `object`. `Any` would also work, but the spirit of PEP 484 is that there shouldn't be `Any` types in fully and precisely annotated programs. We don't want to silently produce an `Any` type if a programmer mistypes a string literal.

A union that includes `object` as an item is equivalent to `object` (it can contain arbitrary objects). Mypy sometimes simplifies these unions and it's okay.

Reply:
Thanks for the correction!